core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -407,13 +408,23 @@ public CreateIndexRequest source(Map<String, ?> source) {
                     } catch (IOException e) {
                         throw new ElasticsearchParseException("failed to parse custom metadata for [{}]", name);
                     }
+                } else {
+                    // found a key which is neither custom defined nor one of the supported ones
+                    if (unsupportedKey == null) {
```

Comment:
The behavior of the create index request is that iff none of the supported keys are found (`settings`, `mappings`, `aliases`) then the top level elements are treated as `settings`.

In other words 
`PUT test
{
    "index.number_of_shards" : 3
}`
would actually create the index `test` with 3 shards.

So if we simply throw here, the preexisting behavior will be altered. And I am reluctant to do so, as there could be users relying on this behavior... (arguably wrong but preexisting)